prime-router/src/main/kotlin/azure/HistoryFunctions.kt

```diff
@@ -290,40 +296,47 @@ open class BaseHistoryFunction : Logging {
                 OffsetDateTime.now().minusDays(DAYS_TO_SHOW),
                 organizationName ?: authClaims.organization.name
             )
-
             @Suppress("NEW_INFERENCE_NO_INFORMATION_FOR_PARAMETER")
-            val reports = headers.sortedByDescending { it.createdAt }.map {
-                var facilities = arrayListOf<Facility>()
-                if (it.bodyFormat == "CSV")
-                    try {
-                        facilities = getFieldSummaryForReportId(
-                            arrayOf("Testing_lab_name", "Testing_lab_CLIA"), it.reportId.toString(), authClaims
-                        )
-                    } catch (ex: Exception) {
-                        // context.logger.info( "Exception during getFieldSummaryForReportId - TestingLabName was not found - no facilities data will be published" );
-                    }
-                val actions = getActionsForReportId(it.reportId.toString(), authClaims)
-
-                val header = workflowEngine.fetchHeader(it.reportId, authClaims.organization)
-
-                val content = if (header.content !== null) String(header.content) else ""
-                val filename = Report.formExternalFilename(header)
-                val mimeType = Report.Format.safeValueOf(header.reportFile.bodyFormat).mimeType
+            val reports = headers.sortedByDescending { it.createdAt }.mapNotNull {
+                val facilities = arrayListOf<Facility>()
+                val actions = arrayListOf<Action>()
+                // get the org passed in
+                val adminOrg = workflowEngine.settings.organizations.firstOrNull { org ->
+                    org.name.lowercase() == organizationName
+                }
+                val header = try {
+                    workflowEngine.fetchHeader(it.reportId, adminOrg ?: authClaims.organization)
```

Comment:
I'm wondering about this fetch and whether its redundant with the call to fetchDownloadableReportFiles, above.   Are we fetching the same data twice?